src/test/lambda/config/templates.test.ts

```diff
@@ -756,3 +763,85 @@ describe('TemplatesConfigPopulator', async () => {
         })
     })
 })
+
+describe('getExistingConfiguration', async () => {
+    let registry: CloudFormationTemplateRegistry
+    let tempFolder: string
+    let tempTemplateFile: vscode.Uri
+    let tempConfigFile: string
+    let fakeWorkspaceFolder: vscode.WorkspaceFolder
+    let tempConfigFolder: string
+    const matchedHandler = "it'sTheSameHandler"
+
+    beforeEach(async () => {
+        tempFolder = await makeTemporaryToolkitFolder()
```

Comment:
Consider creating an abstraction for the filesystem interactions as you would likely do if this were a web service call. This makes it easier to write unit tests that don't cross boundaries (core logic, UI, network, filesystem, process, etc).

There is a lot of setup code here to create the folder, add files, remove the file, populate the registry etc. This can make the tests harder to maintain.

This falls into the category of an integration test. For integration tests, it's generally considered best practice to create reusable utilities for getting the system into the proper state. Consider factoring out the reusable components (like setting up the sample yaml and template registry) into a utility module similar to what's already been done with `makeTemporaryToolkitFolder` and others.

Here, I would suggest a utility to get a fully-populated `CloudFormationTemplateRegistry` or possibly just to create the appropriate files that let the `CloudFormationTemplateRegistry` instantiate successfully

Reply:
> There is a lot of setup code here to create the folder, add files, remove the file, populate the registry etc. This can make the tests harder to maintain.

Mocks add a greater maintenance cost. Not in favor of elaborate mocks just to avoid using the filesystem. These tests will be more maintainable and more useful if they use the real application codepaths.

Reply:
Yes, I wasn't directly suggesting faking out the file system (although I believe you should in unit tests). My suggestion here is more along the lines of encapsulating interactions between the core logic and other boundaries like filesystem access, UI, and service calls. It makes it easier to partition these different components (and fake them if necessary).

My suggestion would be to wrap direct files/filesystem access in some way. This gives the developer options when it comes to satisfying this module's interfaces. It decouples the module from the filesystem and defines a more precise set of operations / data structures that the module operates on.

Reply:
> This falls into the category of an integration test.

FWIW we do this fairly often throughout our tests. We can move this to an integration test, but we may want to just take a larger sweep and move all at once.

Reply:
the `integrationTest/` folder is for things that spin up a full extension and may hit the network. the `test/` folder holds "pure" unit tests and "impure" tests that don't hit the network, but often use the vscode API. Unless this test is hitting the network it should stay in the `test/` folder.

We may later shuffle things around but it's super low priority at the moment (and low-leverage permanently).

Reply:
The disadvantage of keeping them together is that it's hard to tell, as a reviewer and someone reading the code, the intent and coverage of a test. 

i.e. "It this intended to test these 3 modules or just the 1?"

Reply:
The distinction doesn't matter. The tests in `test/` run quickly and off-network, and that's what matters.